spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/session/SessionStoreType.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.session;
+
+/**
+ * Supported Spring Session data store types.
+ *
+ * @author Tommy Ludwig
+ * @since 1.4.0
+ */
+public enum SessionStoreType {
+
+	/**
+	 * Redis backed sessions.
+	 */
+	REDIS,
+
+	/**
+	 * Hazelcast backed sessions.
+	 */
+	HAZELCAST,
+
+	/**
+	 * Simple in-memory map of sessions.
+	 */
+	SIMPLE,
```

Comment:
@snicoll I do not think `SIMPLE` is ideal for a few reasons:

* It does not align with the other configuration options. We didn't name the Redis configuration `SLIGHTLY_TRICKY`. Why would I call the HashMap implementation `SIMPLE`?
* It does not give a clear indication what the implementation is. What do we do when we have other `SIMPLE` options?
* `SIMPLE` is subjective.